src/physical_clock.c

```diff
@@ -0,0 +1,58 @@
+#include "reactor-uc/physical_clock.h"
+#include "reactor-uc/logging.h"
+
+// FIXME: Think about how to handle the case where the clock is stepped back.
+// FIXME: How to handle the case where the clock is stepped forward.
+// FIXME: How to handle concurrency?
```

Comment:
Document this well. The only problem is between async context calling `get_time` and runtime context calling `set_time` or `adjust_time`. So it means that we should only call these things from critical sections.